src/graphics/color.js

```diff
@@ -59,17 +59,18 @@ Crafty.extend({
         }
 
         function parseHexString(hex, c) {
-            var l;
-            if (hex.length === 7){
-                l=2;
-            } else if (hex.length === 4){
-                l=1;
-            } else {
+            if (hex.length === 4){
+                var r = hex.substr(1, 1);
+                var g = hex.substr(2, 1);
+                var b = hex.substr(3, 1);
+                hex = '#'+r+r+g+g+b+b;
+            }
+            if (hex.length !== 7) {
                 return default_value(c);
             }
-            c._red = parseInt(hex.substr(1, l), 16);
-            c._green = parseInt(hex.substr(1+l, l), 16);
-            c._blue = parseInt(hex.substr(1+2*l, l), 16);
+            c._red = parseInt(hex.substr(1, 2), 16);
+            c._green = parseInt(hex.substr(3, 2), 16);
+            c._blue = parseInt(hex.substr(5, 2), 16);
             return c;
         }
```

Comment:
You could probably optimize this code slightly by removing redundant calls to `substr`:
```js
function parseHexString(hex, c) {
    var r, g, b,
        l = hex.length;

    if (l === 7) {
        r = hex.substr(1, 2);
        g = hex.substr(3, 2);
        b = hex.substr(5, 2);
    } else if (l === 4) {
        r = hex.substr(1, 1); r += r;
        g = hex.substr(2, 1); g += g;
        b = hex.substr(3, 1); b += b;
    } else {
        return default_value(c);
    }
    c._red = parseInt(r, 16);
    c._green = parseInt(g, 16);
    c._blue = parseInt(b, 16);

    return c;
}
```